src/components/display/components/CheckboxComponent.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { CheckboxArgs } from 'display-api'
+import { ActivityIndicator, Checkbox, Text } from 'react-native-paper'
+import { StyleSheet, View } from 'react-native'
+import colors from '../../../utils/colors'
+import { globalStyles } from '../../../utils/styles'
+import { ComponentDelegate } from '../ComponentDelegate'
+import { blockMethod, setMethodVariable } from '../method'
+
+export const CheckboxComponent = ({
+	color_scheme,
+	color,
+	variant,
+	readonly,
+	disabled,
+	name,
+	value,
+	text,
+	on_change,
+}: CheckboxArgs) => {
+	color = color || colors.text
+
+	const [checked, setChecked] = React.useState(value)
+	const [isLoading, setLoading] = React.useState(false)
+
+	const onPress = async () => {
+		if (variant === 'Default') {
+			setChecked(checked === 0 ? 1 : 0)
+		} else {
+			// TODO: handle the cancel check state
```

Comment:
I won't close the checkbox ticket without handling the cancel check state, but it's not a priority if it gets spilled to other sprints.

Reply:
Can you please elaborate on what would the cancel state look like on the checkbox UI wise?

Reply:
Either an "X" or a dash "-", gray-ed out.
<img width="370" alt="Screen Shot 2021-03-15 at 14 25 56" src="https://user-images.githubusercontent.com/32454680/111209707-5e63f900-859a-11eb-92a8-ed2826c88d66.png">